src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1485,7 +1485,7 @@ private void ExecuteAndCatchIfThrows_NoLock<TArg>(DiagnosticAnalyzer analyzer, A
 
         internal bool ExceptionFilter(Exception ex)
         {
-            if ((ex as OperationCanceledException)?.CancellationToken == _cancellationToken)
+            if (ex is OperationCanceledException { CancellationToken: _cancellationToken })
```

Comment:
>_cancellationToken [](start = 70, length = 18)

Not a valid pattern

Reply:
➡️ Treating this one as a duplicate of https://github.com/dotnet/roslyn/pull/39738#discussion_r344405091